### DISCUSSION TITLE
Squib Helper Utilities

### DISCUSSION THREAD

**Discussion by rnewton** [andymeneely/squib#350] [Show and tell]
I've created a new gem that I've found helpful and maybe others will as well. None of these things felt like they should be in Squib proper, but I use them frequently. [Check it out here](https://github.com/coinflipgamesllc/squib_plus)

**Comment by andymeneely** [comment]
Awesome!! I’ll take a look.

**Comment by andymeneely** [comment]
Google sheets lol. Yeah long ago I got burned in another project by Sheets suddenly changing its APIs. Ironically they haven’t in recent years. 

I like the merge pdf! I do something else for front/back but your way is waaay simpler. 

String monkey patch - I actually have some similar things myself. I implemented them as Ruby refinements so they won’t collide. Haven’t put them in Squib (yet) but maybe I should. I’d be up for adding slug. 

I like the card num method. I like convenience methods like that. Same with tts - I’d be ok adding that to Squib. 

Card sizes is cool! Great idea. 

LOVE the avatar idea! And it warms my heart that you added the new dsl method using the same technique I do with accepted_params and args - I’m proud of that design. 

Yeah most of that stuff I’d be up for adding into Squib if you’d like. Or we can keep it in squib-plus. Up to you.

**Reply by rnewton** [reply]
Happy to make a PR or two for merging these into squib if you like em. I'll add some tests and documentation as well

**Reply by andymeneely** [reply]
Go for it! Thanks!

Be sure to add some samples to the samples folder. Then take any images you want me to regression test and add “_expected” to the end. It’ll get put into my visual regression tests. You can test it out with “rake sanity”
